packages/suite/src/views/wallet/send/components/Outputs/components/Address/index.tsx

```diff
@@ -198,9 +206,56 @@ export const Address = ({ output, outputId, outputsCount }: AddressProps) => {
                 }
             },
             // eth addresses are valid without checksum but Trezor displays them as checksummed
-            checksum: (value: string) => {
-                if (networkType === 'ethereum' && !checkAddressCheckSum(value)) {
-                    return translationString('RECIPIENT_IS_NOT_VALID');
+            used_checksum: async (value: string) => {
+                let lastTwoValues = JSON.parse(localStorage.getItem('lastTwoValues') || '[]');
+                lastTwoValues.push(value);
+                if (lastTwoValues.length > 2) {
+                    lastTwoValues.shift();
+                }
+                localStorage.setItem('lastTwoValues', JSON.stringify(lastTwoValues));
+                //they are not the same but the second one is checksum of the first one, in this case show the info that the address was automatically checksummed
+                if (
+                    lastTwoValues[0] !== lastTwoValues[1] &&
+                    toChecksumAddress(lastTwoValues[0]) == lastTwoValues[1]
+                ) {
+                    setAddressInfoContent({
+                        text: 'Address converted to checksum format.',
+                    });
+                } else {
+                    setAddressInfoContent(undefined);
+                }
+
+                //1.If the address is used but unchecksummed or incorrectly checksummed, then Suite will automatically convert the address to the correct checksummed form and inform the user as described in the OP.
+                if (networkType === 'ethereum') {
+                    const params = {
+                        descriptor: value,
+                        coin: 'ethereum',
+                    };
+                    const result = await TrezorConnect.getAccountInfo(params); //TODO handle error
+                    if (result.success && result.payload.history.total !== 0) {
+                        if (!checkAddressCheckSum(value)) {
+                            console.log('used but not checksummed');
+                            setValue(inputName, toChecksumAddress(value), {
+                                shouldValidate: true,
+                            });
+                        }
+                    }
+                }
+            },
+            checksum: async (value: string) => {
+                if (networkType === 'ethereum') {
+                    const params = {
+                        descriptor: value,
+                        coin: 'ethereum',
+                    };
+                    const result = await TrezorConnect.getAccountInfo(params);
+                    //2. if address is not checksummed at all and not found in blockbook
+                    //offer to checksum width a button
+                    if (result.success && result.payload.history.total == 0) {
+                        if (value == value.toLowerCase()) {
```

Comment:
```suggestion
                        if (address === address.toLowerCase()) {
```
what if it is not checksummed nor lowercase? e.g. its all uppercase 😀  or mixed case but not in the correct checksummed way

Reply:
If it's all uppercase, it can be treated the same way as all lowercase. But in case of mixed case (but wrong checkdummed), it should be just invalid, because it means that the address was affected by a typo. By automatically checksumming such an address we put the user under a risk of using a non-existent address. Hence the user should double-check and fix it himself.

Reply:
If it's incorrectly checksummed in mixed case it's catched much earlier than the checksum validation. So no matter if the account is used, the uncorrectly checksummed address will be not valid. Here: 

            valid: (value: string) => {
                if (!isAddressValid(value, symbol)) {
                    return translationString('RECIPIENT_IS_NOT_VALID');
                }
            },

Reply:
If it's all upprercase it's a bit tricky. If the x in 0x is uppercase too, it's invalid, if the x is lowercase but the rest of the address is all uppercase, it gets converted to checksum. Is that the desired behavior?